pkg/api/types.go

```diff
@@ -2303,8 +2303,9 @@ type FinalizerName string
 
 // These are internal finalizer values to Kubernetes, must be qualified name unless defined here
 const (
-	FinalizerKubernetes FinalizerName = "kubernetes"
-	FinalizerOrphan     string        = "orphan"
+	FinalizerKubernetes    FinalizerName = "kubernetes"
+	FinalizerOrphan        string        = "orphan"
+	FinalizerSynchronousGC string        = "collectingGarbage"
```

Comment:
I can change this new finalizer to CamelCase, but not the old ones, otherwise the finalizers won't work accross cluster upgrade.